src/test/groovy/com/github/eirnym/js2p/JavaTaskFunctionalTest.groovy

```diff
@@ -117,4 +114,44 @@ class JavaTaskFunctionalTest {
         Files.copy(Paths.get("demo", "java", "src", "main", "resources", "json", "address.json"), addressJson)
     }
 
+    private static void assertExists(File file) {
+        assertNotNull(file)
+        assertTrue(file.exists())
+    }
+
+    private static BuildResult executeRunner(String gradleVersion, Path testProjectDir, String task = COLON_TASK_NAME) {
+        def arguments = GradleRunner.create()
+                .withPluginClasspath()
+                .withProjectDir(testProjectDir.toFile())
+                .withArguments(task, "--info")
+
+        if (gradleVersion) {
+            arguments.withGradleVersion(gradleVersion)
+        }
+        return arguments.build()
+    }
+
+    static boolean gradleSupported(String gradleVersion, String javaSpecificationVersion) {
+        if (!gradleVersion || javaSpecificationVersion == '1.8' || javaSpecificationVersion.toInteger() < 13) {
+            return true
+        }
+
+        def parts = gradleVersion.split(/\./)
+
+        if (parts[0].toInteger() < 6) {
+            return false
+        }
+        switch (javaSpecificationVersion.toInteger()) {
+            case 12: return true  // no information on this version of Java?
+            case 13: return true
+            case 14: return parts[1].toInteger() >= 3
+            case 15: return parts[1].toInteger() >= 6
+        }
+    }
```

Comment:
Not all execution paths return a value.
Please add a `default: return false`

Reply:
Fixed and comment was added